### PR TITLE
Support multiple base AMI channels

### PR DESCRIPTION
**Pull request by aermakov-zalando**
- config.base_ami: rename name to default_channel, add an optional channels list
 - only the latest base AMI is configured for each region and channel
 - list of latest base AMIs is prepared once and then reused across all accounts to avoid rate limit issues. per-account overrides are thus unsupported, but I don't think they make a lot of sense
 - dropped an unused configure_bastion_host_deprecated function

### CODE REVIEW

File: sevenseconds/cli.py [hunk 1]
```diff
@@ -80,11 +80,12 @@ def configure(file, account_name_pattern, **options):
             file,
             account_name_pattern,
             options)
+        sessions = list(sessions.values())
```
Comment (tuxlife) [inline sevenseconds/cli.py]: I find it ugly that the variable is initialized as hash and then overwritten as an array

File: sevenseconds/config/__init__.py [hunk 2]
```diff
@@ -39,28 +40,58 @@
         'auth'              # OAuthServices Object (exp. for Account List and AWS Credentials Service)
     ))
 
+SharedData = namedtuple(
```
Comment (tuxlife) [inline sevenseconds/config/__init__.py:43]: Why did you use SharedData? Lastest AMI isn't the same for all accounts.

File: examples/example-full-configuration.yaml [hunk 3]
```diff
@@ -23,7 +23,10 @@ global:
   saml_admin_login_role: "Shibboleth-Administrator"
   # base Taupage AMI to search for
   base_ami:
-    name: "Taupage-AMI-*"
+    default_channel: "Taupage-AMI-*"
+    channels:
```
Comment (tuxlife) [inline examples/example-full-configuration.yaml:27]: why did you need the channels in sevenseconds? If one account need a other taupage app then you can set the base_ami for this account

```
foobar:
  base_ami:
    name: "TaupageStaging-AMI-*"
```

**Review by tuxlife — CHANGES_REQUESTED** [review]

File: examples/example-full-configuration.yaml (hunk 3 above)
Comment (aermakov-zalando) [reply examples/example-full-configuration.yaml:27]: To correctly setup Taupage for AWS accounts, we either share the image during build time (for existing accounts), or use sevenseconds on new accounts which deals with it.
Unfortunately, the logic it uses doesn't really scale. It looks for one existing image (`Taupage-AMI-…`) and only if that isn't present shares every image. Moreover, there's no explicit step for this, it's just a side effect of bastion host/VPC initalization. This means that 1) it's impossible to automatically share staging AMIs and 2) if there's any issues the account might be stuck in a weird state.
I've changed the logic to be a lot more straightforward: we define explicit channels in the configuration, fetch the latest AMIs for each channel and then share the AMIs with the accounts if they are not shared. The latest AMI fetch is also done once to avoid bumping into rate limits, since I don't think overriding the Taupage configuration for an account is something that we ever needed, and there's already some parts of sevenseconds that are done globally.

File: sevenseconds/config/__init__.py (hunk 2 above)
Comment (aermakov-zalando) [reply sevenseconds/config/__init__.py:43]: See the comment above. In the new model it is, because with the staging channel we don't need the manual sharing step anymore.

**Comment by aermakov-zalando** [comment]
@tuxlife asked to move the Taupage sharing logic from sevenseconds to aws-account-creator, so I'm closing this.

**Comment by aermakov-zalando** [comment]
Let's discuss this again. I've had a look at the aws-account-creator code and right now it does absolutely nothing other than creating the account and then running sevenseconds several times to actually initialise it. While I agree that it might make sense to move most of the logic there, I think it should be done as part of a separate issue, not as part of changing one part of the logic.

**Comment by szuecs** [comment]
@tuxlife for me this looks good and as discussed in our meeting, it's a change from the old share based model to a channel based, that works pretty fine in our cluster setup. I think in general it makes a lot of sense.
I also think it makes sense that this change lives in sevenseconds. Maybe we have to define the scope from our internal tool and sevenseconds, such that it is more clear what to do where.
From my side lgtm

**Review by tuxlife — APPROVED** [review]

**Comment by aermakov-zalando** [comment]
👍

**Comment by mikkeloscar** [comment]
:+1:
